File: CS/GrowthWare.Framework.BusinessData/DataAccessLayer/Interfaces/IDBInformation.cs
using GrowthWare.Framework.BusinessData.DataAccessLayer.Interfaces.Base;
using GrowthWare.Framework.Model.Profiles;
using System.Data;

namespace GrowthWare.Framework.BusinessData.DataAccessLayer.Interfaces
{
    public interface IDBInformation : IDDBInteraction
    {
        MDBInformation Profile
        {
            get;
            set;

        }
        DataRow GetProfile();
        bool UpdateProfile();
    }
}

File: VB/GrowthWare.Framework.BusinessData/DataAccessLayer/Interfaces/IDBInformation.vb
Imports GrowthWare.Framework.BusinessData.DataAccessLayer.Interfaces.Base
Imports GrowthWare.Framework.Model.Profiles

Namespace DataAccessLayer.Interfaces
    ''' <summary>
    ''' Public interface for DDBInformation
    ''' </summary>
    Public Interface IDBInformation
        Inherits IDDBInteraction

        ''' <summary>
        ''' Gets or sets the profile.
        ''' </summary>
        ''' <value>The profile.</value>
        Property Profile As MDBInformation

        ''' <summary>
        ''' Gets the profile.
        ''' </summary>
        ''' <returns>DataRow.</returns>
        ReadOnly Property GetProfileRow() As DataRow

        ''' <summary>
        ''' Updates the profile.
        ''' </summary>
        ''' <returns><c>true</c> if XXXX, <c>false</c> otherwise.</returns>
        Function UpdateProfile() As Boolean
    End Interface

End Namespace